src/Helpers/ApartmentHelpers.php

```diff
@@ -0,0 +1,44 @@
+<?php
+
+namespace BuildEmpire\Apartment\Helpers;
+
+class ApartmentHelpers
+{
+    /**
+     * Does the schema only contain letters and/or numbers.
+     *
+     * @param $schemaName
+     * @return bool
+     */
+    public static function isSchemaNameValid($schemaName)
+    {
+        return (preg_match('/^[a-z][a-z0-9_]*$/', $schemaName) && !ctype_upper($schemaName));
+    }
+
+    /**
+     * This will remove any bad characters from the $schemaName variable.
+     *
+     * Note: Removes any non valid strings with #. You should be checking with isSchemaNameValid before reaching this
+     * section. This is just an extra layer of protection.
+     *
+     * @param $schemaName
+     * @return mixed
+     */
+    public static function getSchemaSafeString($schemaName) {
+        return preg_replace('/^[^0-9a-z_]+$/', '#',$schemaName);
```

Comment:
No, it would fail. It would throw an statement exception. I agree, this is not an ideal solution but the reason for doing this really are:

1. Anyone writing/extending Apartment in the future should be checking the data passing in is correct, which all our methods should be doing.

2. It's a last sort of attempt to stop anything bad being passed into the statement.

It's not really required as we are checking the data before it's passed but just an extra layer of security.